File: ddmlib/src/main/java/com/android/ddmlib/input/InputDispatcher.java
package com.android.ddmlib.input;

import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.TimeUnit;

/**
 * Created by majipeng on 2017/6/19.
 */
public class InputDispatcher {
    private BlockingQueue<MonitorEvent> waitingForDispatch = new LinkedBlockingQueue<>(2);

    public InputDispatcher(InputManager inputManager) {
    }

    void dispatchEvent(MonitorEvent event) {
        waitingForDispatch.add(event);
    }

    MonitorEvent getWaitingForDispatchEvent() throws InterruptedException {
        return waitingForDispatch.poll(200, TimeUnit.MILLISECONDS);
    }
}

File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputDispatcher.kt
package com.android.ddmlib.input.android

import java.util.concurrent.BlockingQueue
import java.util.concurrent.LinkedBlockingQueue

/**
 * Created by majipeng on 2017/6/19.
 */
class InputDispatcher(inputManager: InputManager) {
    private val waitingForDispatch = LinkedBlockingQueue<MonitorEvent>(2)

    internal fun dispatchEvent(event: MonitorEvent) {
        waitingForDispatch.add(event)
    }

     val waitingForDispatchEvent: MonitorEvent
        @Throws(InterruptedException::class)
        get() = waitingForDispatch.take()
}
